src/managers/GuildInviteManager.js

```diff
@@ -0,0 +1,179 @@
+   * @example
+   * // Fetch all invites from a guild
+   * guild.invites.fetch()
+   *   .then(console.log)
+   *   .catch(console.error);
+   * @example
+   * // Fetch all invites from a guild without caching
+   * guild.invites.fetch({ cache: false })
+   *   .then(console.log)
+   *   .catch(console.error);
+   * @example
+   * // Fetch a single invite
+   * guild.invites.fetch('DrzKVU3')
+   *   .then(console.log)
+   *   .catch(console.error);
+   * @example
+   * // Fetch a single invite without checking cache
+   * guild.invites.fetch({ code: 'DrzKVU3', force: true })
+   *   .then(console.log)
+   *   .catch(console.error)
+   * @example
+   * // Fetch a single invite without caching
+   * guild.invites.fetch({ code: 'DrzKVU3', cache: false })
+   *   .then(console.log)
+   *   .catch(console.error);
+   */
+  fetch(options) {
+    if (!options) return this._fetchMany();
+    const code = DataResolver.resolveInviteCode(options);
+    if (code) return this._fetchSingle({ code, cache: true });
+    if (options.code) {
+      options.code = DataResolver.resolveInviteCode(options);
+    }
+    if (!options.code) {
+      if ('cache' in options) return this._fetchMany(options.cache);
+      return Promise.reject(new Error('INVITE_RESOLVE_CODE'));
+    }
+    return this._fetchSingle(options);
+  }
+
+  async _fetchSingle({ code, cache, force = false }) {
+    if (!force) {
+      const existing = this.cache.get(code);
+      if (existing) return existing;
+    }
+
+    const invites = await this._fetchMany(cache);
+    const invite = invites.get(code);
+    if (!invite) throw new Error('UNKNOWN_INVITE');
+    return invite;
+  }
+
+  async _fetchMany(cache) {
+    const data = await this.client.api.guilds(this.guild.id).invites.get();
+    return data.reduce((col, invite) => col.set(invite.code, this.add(invite, cache)), new Collection());
+  }
+
+  /**
+   * Create a invite to the guild from a random channel.
+   * @param {CreateInviteOptions|ChannelResolvable} [options=null] The options for creating the invite
+   * @param {ChannelResolvable} [channel=null] The options for creating the invite
+   * @returns {Promise<Invite?>}
+   * @example
+   * // Create an invite to a random allowed channel
+   * guild.invites.create()
+   *   .then(invite => console.log(`Created an invite with a code of ${invite.code} from ${invite.channel}`))
+   *   .catch(console.error);
+   * @example
+   * // Create an invite to a selected channel
+   * guild.invites.create('599942732013764608')
+   *   .then(console.log)
+   *   .catch(console.error);
+   */
+  create(options = null, channel = null) {
+    if (!channel && (options instanceof GuildChannel || typeof options === 'string')) {
+      channel = options;
+      options = null;
+    }
+    if (!channel) {
+      channel = this.guild.channels.cache.find(c => c.permissionsFor(this.guild.me).has('CREATE_INSTANT_INVITE'));
+      if (!channel) throw new Error('GUILD_CHANNEL_INVITE');
```

Comment:
Either make this entire function async, or wrap these in return Promise.reject, as right now these are sync errors